### PR TITLE
The Real Travis CI additions. 

### PR DESCRIPTION
**Pull request by opotowsky**
This reverts commit d779a15925860a898d2b91e1c75ef322b692f675.

Getting back on track, see this conversation: https://github.com/cyclus/cymetric/pull/72

### CODE REVIEW

**Comment by scopatz** [comment]
HI @opotowsky - it looks like this build fails...

**Comment by gidden** [comment]
Hi @scopatz, @opotowsky has noted the failure and her (and my) confusion a
few times [1, 2], most recently [1].

[1] https://github.com/cyclus/cymetric/pull/72#issuecomment-101024601
[2] https://github.com/opotowsky/cymetric/pull/2#discussion_r29907484

On Mon, May 11, 2015 at 8:43 PM, Anthony Scopatz <notifications@github.com>
wrote:

> HI @opotowsky <https://github.com/opotowsky> - it looks like this build
> fails...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/cymetric/pull/73#issuecomment-101095354>.
>



-- 
Matthew Gidden, Ph.D.
Postdoctoral Associate, Nuclear Engineering
The University of Wisconsin -- Madison
Ph. 225.892.3192

**Comment by scopatz** [comment]
Two thoughts:

1. The asmeurer channel is not added here, though I believe that it is in the condarc.

2. Some of these dependencies don't match up with cyclus's on that platform.  For example cyclus does not need a glib on linux but does require our custom myglib on mac.  These dependencies are probably not even really needed anymore now that we are not on batlab, or at least our custom versions aren't.

I'd start out by seeing if 1 fixes it, and if not, then play around with 2.

**Comment by gidden** [comment]
The condarc from ciclus 0.1 was:
```
$ cat condarc 
# This is a sample .condarc file

# channel locations. These override conda defaults, i.e., conda will
# search *only* the channels listed here, in the order given. Use "defaults" to
# automatically include all default channels. Non-url channels will be
# interpreted as binstar usernames (this can be changed by modifying the
# channel_alias key; see below).
channels:
  - https://conda.binstar.org/cyclus
  - defaults
```

**Comment by gidden** [comment]
Also, note the diffs in travis.ymls:
```
$ diff cycamore/.travis.yml cymetric/.travis.yml
33c33
<   - cd ../cycamore
---
>   - cd ../cymetric
35c35
<   # install cycamore  
---
>   # install cymetric
37c37
< 
---
>   
41,42d40
<   - cycamore_unit_tests
<   - conda install numpy pytables
```

**Comment by gidden** [comment]
So @scopatz, I agree that the meta.yaml could definitely be cleaned up. I didn't realize that it was so much more verbose than cycamore's.

However, @opotowsky, it looks like this is a file permissions issue. Try doing the following `chmod 755 .travis-install.sh` and commit/push that.

**Comment by opotowsky** [comment]
@scopatz this is the most recent failure. https://travis-ci.org/cyclus/cymetric/jobs/62255548#L2207

I cleaned up the cymetric meta.yaml file to delete all overlap with the cyclus meta.yaml.

**Comment by scopatz** [comment]
This is because the actual build of cymetric failed silently: https://travis-ci.org/cyclus/cymetric/jobs/62255548#L2093

**Comment by gidden** [comment]
probably need travis c++11 support. from googling:
- http://stackoverflow.com/questions/22111549/travis-ci-with-clang-3-4-and-c11
- http://gearchicken.com/wordpress/?p=105

**Comment by opotowsky** [comment]
Thanks @gidden; trying now

**Comment by opotowsky** [comment]
ping @gidden and @scopatz, I think we are making progress. Do I need to re-add the cycamore build in .travis.yml? https://travis-ci.org/cyclus/cymetric/jobs/62299276#L3582

**Comment by gidden** [comment]
It appears that the cymetric unit tests depend on cycamore. In the short term -- yes, you will have to add back the cycamore build. In the long term, this likely shouldn't be the case... (from previous discussion the dependency tree should be cyclus->cymetric->cycamore [eventually]).

These other test failures are real, though, and I have no idea where they come from. This will likely need to be investigated by you or @scopatz. Here is an output of my unit test run without cycamore installed: https://gist.github.com/gidden/5ec5d3680d464e438cd6

**Comment by scopatz** [comment]
All of these look like they come from the fact that the simulations didn't successfully run.

**Comment by gidden** [comment]
The `test-input.xml` file should be updated for 1.3

**Comment by opotowsky** [comment]
Test failures (and two skips, possibly something pyne related?):
1. Known fix here, my guess is a merging error is the reason pyne.enrichment isn't in the file. https://travis-ci.org/opotowsky/cymetric/jobs/62323516#L4673
2. I'm not sure about why this is failing here, but passes locally. Should I use DataFrame.eval() instead? https://travis-ci.org/opotowsky/cymetric/jobs/62323516#L4693
3. No idea! Everything looks right to me but this is failing. https://travis-ci.org/opotowsky/cymetric/jobs/62323516#L4703

**Comment by gidden** [comment]
On Tue, May 12, 2015 at 10:03 PM, opotowsky <notifications@github.com>
wrote:

> Test failures (and two skips, possibly something pyne related?):
> 1. Known fix here, my guess is a merging error is the reason
> pyne.enrichment isn't in the file.
> https://travis-ci.org/opotowsky/cymetric/jobs/62323516#L4673
>
You probably need to add pyne as a dependency in the meta.yaml (I don't
think it is documented anywhere as a dependency and probably should be).
Separately, fco_u_mined should probably have a `raise_no_pyne` call at the
beginning like other pyne-related functions.

>
> 2. I'm not sure about why this is failing here, but passes locally. Should
> I use DataFrame.eval() instead?
> https://travis-ci.org/opotowsky/cymetric/jobs/62323516#L4693
>
numexpr should probably be added as a dependency.

>
> 3. No idea! Everything looks right to me but this is failing.
> https://travis-ci.org/opotowsky/cymetric/jobs/62323516#L4703
>
Yeah, I also don't know. Maybe you'll get lucky and this will go away with
the first two fixed? Otherwise maybe @scopatz can chime in.

> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/cymetric/pull/73#issuecomment-101492293>.
>



-- 
Matthew Gidden, Ph.D.
Postdoctoral Associate, Nuclear Engineering
The University of Wisconsin -- Madison
Ph. 225.892.3192

**Comment by scopatz** [comment]
> You probably need to add pyne as a dependency in the meta.yaml 

PyNE should be an optional dependency, though it should probably also be installed.

> numexpr should probably be added as a dependency.

If you add pyne, numexpr will come along for the ride.

> Otherwise maybe @scopatz can chime in.

This is probably because the wrong version of the code is being installed or used or maybe a variable name has been changed.

**Comment by gidden** [comment]
On Wed, May 13, 2015 at 9:42 AM, Anthony Scopatz <notifications@github.com>
wrote:

> You probably need to add pyne as a dependency in the meta.yaml
>
> PyNE should be an optional dependency, though it should probably also be
> installed.
>
> numexpr should probably be added as a dependency.
>
> If you add pyne, numexpr will come along for the ride.
>
 It looks like without pyne it is still needed (at least for the tests).

>  Otherwise maybe @scopatz <https://github.com/scopatz> can chime in.
>
> This is probably because the wrong version of the code is being installed
> or used or maybe a variable name has been changed.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/cymetric/pull/73#issuecomment-101692068>.
>



-- 
Matthew Gidden, Ph.D.
Postdoctoral Associate, Nuclear Engineering
The University of Wisconsin -- Madison
Ph. 225.892.3192

**Comment by scopatz** [comment]
> It looks like without pyne it is still needed (at least for the tests).

Is pyne being installed for the test phase?

**Comment by gidden** [comment]
On Wed, May 13, 2015 at 9:46 AM, Anthony Scopatz <notifications@github.com>
wrote:

> It looks like without pyne it is still needed (at least for the tests).
>
> Is pyne being installed for the test phase?
>
Not yet, but I see your point.

> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/cymetric/pull/73#issuecomment-101695351>.
>



-- 
Matthew Gidden, Ph.D.
Postdoctoral Associate, Nuclear Engineering
The University of Wisconsin -- Madison
Ph. 225.892.3192

**Comment by opotowsky** [comment]
So I added pyne to the test requirements but I think all the tests that need pyne are still skipping, and this one still has the related error https://travis-ci.org/cyclus/cymetric/jobs/62423974#L4689

**Comment by gidden** [comment]
to get more output, update the nosetest line to `nosetest -svw tests`

you can also do a hack to see what's in the metrics.py file in the
installation script by catting metrics.py in .travis.yml

On Wed, May 13, 2015 at 12:10 PM, opotowsky <notifications@github.com>
wrote:

> So I added pyne to the test requirements but I think all the tests that
> need pyne are still skipping, and this one still has the related error
> https://travis-ci.org/cyclus/cymetric/jobs/62423974#L4689
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/cymetric/pull/73#issuecomment-101749273>.
>



-- 
Matthew Gidden, Ph.D.
Postdoctoral Associate, Nuclear Engineering
The University of Wisconsin -- Madison
Ph. 225.892.3192

**Comment by opotowsky** [comment]
Thanks @gidden!

**Comment by gidden** [comment]
You might be able to get around the pyne issue by explicitly `conda install
pyne` in the script portion of .travis.yml. Though I do not know why what
you have isn't working.

On Wed, May 13, 2015 at 12:14 PM, Matthew Gidden <gidden@wisc.edu> wrote:

> to get more output, update the nosetest line to `nosetest -svw tests`
>
> you can also do a hack to see what's in the metrics.py file in the
> installation script by catting metrics.py in .travis.yml
>
> On Wed, May 13, 2015 at 12:10 PM, opotowsky <notifications@github.com>
> wrote:
>
>> So I added pyne to the test requirements but I think all the tests that
>> need pyne are still skipping, and this one still has the related error
>> https://travis-ci.org/cyclus/cymetric/jobs/62423974#L4689
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/cyclus/cymetric/pull/73#issuecomment-101749273>.
>>
>
>
>
> --
> Matthew Gidden, Ph.D.
> Postdoctoral Associate, Nuclear Engineering
> The University of Wisconsin -- Madison
> Ph. 225.892.3192
>



-- 
Matthew Gidden, Ph.D.
Postdoctoral Associate, Nuclear Engineering
The University of Wisconsin -- Madison
Ph. 225.892.3192

**Comment by opotowsky** [comment]
Should I do that now or wait for this current run to give us output?

**Comment by gidden** [comment]
I would let the current run start then push the commit. You can have two
running simultaneously.

On Wed, May 13, 2015 at 12:24 PM, opotowsky <notifications@github.com>
wrote:

> Should I do that now or wait for this current run to give us output?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/cymetric/pull/73#issuecomment-101752118>.
>



-- 
Matthew Gidden, Ph.D.
Postdoctoral Associate, Nuclear Engineering
The University of Wisconsin -- Madison
Ph. 225.892.3192

**Comment by opotowsky** [comment]
This is extremely confusing but may explain why the test for fco_electricity_generated is also failing:
https://travis-ci.org/cyclus/cymetric/jobs/62436039#L5163. The metric is `FcoUMined` and the column should be `UMined`. This was not always the case. Is this building off a different version of cymetric?

**Comment by gidden** [comment]
Try updating the meta.yaml: ["Travis" not in environ] -> ["TRAVIS" not in
environ]

On Wed, May 13, 2015 at 1:47 PM, opotowsky <notifications@github.com> wrote:

> This is extremely confusing but may explain why the test for
> fco_electricity_generated is also failing:
> https://travis-ci.org/cyclus/cymetric/jobs/62436039#L5163. The metric is
> FcoUMined and the column should be UMined. This was not always the case.
> Is this building off a different version of cymetric?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/cymetric/pull/73#issuecomment-101773775>.
>



-- 
Matthew Gidden, Ph.D.
Postdoctoral Associate, Nuclear Engineering
The University of Wisconsin -- Madison
Ph. 225.892.3192

**Comment by opotowsky** [comment]
Here, Here!

**Comment by opotowsky** [comment]
In addition to the troubleshooting stuff, should I remove `conda install pyne` from .travis.yml?

**Comment by gidden** [comment]
Feel free to try. Just make sure nothing skips afterward. You can also remove the `cat`s

**Comment by opotowsky** [comment]
Ok I will. If pyne isn't found, are we OK with the forced install of pyne?

**Comment by gidden** [comment]
We are ok with whatever works for now that runs all of the tests.

On Wed, May 13, 2015 at 3:13 PM, opotowsky <notifications@github.com> wrote:

> Ok I will. If pyne isn't found, are we OK with the forced install of pyne?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/cymetric/pull/73#issuecomment-101801782>.
>



-- 
Matthew Gidden, Ph.D.
Postdoctoral Associate, Nuclear Engineering
The University of Wisconsin -- Madison
Ph. 225.892.3192

**Comment by opotowsky** [comment]
ping! finally ready. (@gidden @scopatz)

File: conda-recipe/meta.yaml [hunk 1]
```diff
@@ -35,6 +20,12 @@ requirements:
...
 
+test:
+  requires:
+    - cycamore ==0.0
+    - pyne
+    - nose
+
 about:
-  home: Cymetric
+  home: Cyclus
```
Comment (scopatz) [inline conda-recipe/meta.yaml]: This seems wrong...

Comment (opotowsky) [inline conda-recipe/meta.yaml]: Cycamore had Cyclus here so I thought it was supposed to be the project name of Cyclus

Comment (scopatz) [inline conda-recipe/meta.yaml]: Cycamore was wrong too then :)

**Comment by scopatz** [comment]
Seems good. Thanks @opotowsky

**Pull request merged**